api/api-referential/referential-external/src/test/java/fr/gouv/vitamui/referential/external/server/rest/SchemaExternalControllerTest.java

```diff
@@ -0,0 +1,57 @@
+package fr.gouv.vitamui.referential.external.server.rest;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import fr.gouv.vitamui.commons.rest.client.ExternalHttpContext;
+import fr.gouv.vitamui.iam.security.service.ExternalSecurityService;
+import fr.gouv.vitamui.referential.external.server.service.SchemaExternalService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockMultipartFile;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class SchemaExternalControllerTest {
+
+    private SchemaExternalController schemaExternalController;
+    private SchemaExternalService schemaExternalService;
+    private ExternalSecurityService externalSecurityService;
+
+    @BeforeEach
+    public void setUp() {
+        schemaExternalService = mock(SchemaExternalService.class);
+        externalSecurityService = mock(ExternalSecurityService.class);
+        schemaExternalController = new SchemaExternalController(schemaExternalService, externalSecurityService);
+    }
+
+    @Test
+    void testImportSchemas_Success() throws IOException {
+        // Given
+        JsonNode expectedResponse = mock(JsonNode.class);
+
+        // Mock the behavior of schemaExternalService.importSchemas
+        Mockito.when(schemaExternalService.importSchemas(any(), any(), any())).thenReturn(expectedResponse);
+
+        // Mock the externalSecurityService.getHttpContext() to return a non-null ExternalHttpContext
+        ExternalHttpContext mockHttpContext = mock(ExternalHttpContext.class);
+        Mockito.when(externalSecurityService.getHttpContext()).thenReturn(mockHttpContext);
+        Mockito.when(mockHttpContext.getTenantIdentifier()).thenReturn(1);
+
+        // Load a real CSV file from disk
+        byte[] fileContent = Files.readAllBytes(Paths.get("src/test/resources/data/import_schema.csv"));
+        MockMultipartFile validFile = new MockMultipartFile("file", "import_schema.csv", "text/csv", fileContent);
```

Comment:
À vérifier, mais plutôt quelque chose comme ça :
```suggestion
        MultipartFile validFile = new MockMultipartFile("file", "import_schema.csv", "text/csv", getClass().getResourceAsStream("/data/import_schema.csv"));
```

Reply:
Done

Reply:
Ok